### PR TITLE
EWB-4124: Update super pom to 0.34.1

### PR DESCRIPTION
**Pull request by Lax125**
# Description

Use super pom version 0.34.1, fixing transient vulnerabilities from `grpc-netty-shaded:1.57.2`.

# Associated tasks

Tasks blocking this one:
- [x] https://github.com/zepben/evolve-super-pom/pull/36

Tasks this is blocking:
- https://github.com/zepben/hosting-capacity-utils/pull/12

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [x] I have performed a self review of my own code (including checking issues raised when creating the PR).
- [ ] ~I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.~ (Tests did not break with version updates)
- [ ] ~I have commented my code in any hard-to-understand or hacky areas.~
- [x] I have handled all new warnings generated by the compiler or IDE.
- [x] I have rebased onto the target branch (usually main).
      
### Documentation
- [x] I have updated the changelog.
- [ ] ~I have updated any documentation required for these changes.~

# Breaking Changes
- [x] I have considered if this is a breaking change and will communicate it with other team members if so.

No breaking changes.

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>com.zepben.maven</groupId>
         <artifactId>evolve-super-pom</artifactId>
-        <version>0.33.0</version>
+        <version>0.34.1</version>
```
Comment (kgreav) [inline java/pom.xml:8]: You'll need to update the `setup.py` and `setup.cfg` to have the matching version of gRPC

**Review by kgreav — CHANGES_REQUESTED** [review]

File: java/pom.xml (hunk 1 above)
Comment (kgreav) [reply java/pom.xml:8]: Oh and also the C# project please.

Comment (Lax125) [reply java/pom.xml:8]: It looks like the protobuf version in the python project is already matching if the rule is major version +1 (e.g. `4.24.2` in `setup.py` matches `3.24.2` in the maven project). See https://pypi.org/project/protobuf/#history and https://mvnrepository.com/artifact/com.google.protobuf/protobuf-java.

Comment (Lax125) [reply java/pom.xml:8]: As for the grpc versions, they don't match exactly for the pypi package and don't match at all for the nuget package, even when it comes to release dates. See https://mvnrepository.com/artifact/io.grpc/grpc-protobuf, https://pypi.org/project/grpcio/#history, and https://www.nuget.org/packages/Grpc.Core#versions-body-tab. I'll use 1.59.3 for the python project and 2.46.6 for the C#.

**Review by charlta — APPROVED** [review]

**Pull request merged**